Navitia.php

```diff
@@ -2,173 +2,78 @@
 
 namespace CanalTP\NavitiaPhp;
 
-use CanalTP\NavitiaPhp\Api\Coords;
-use CanalTP\NavitiaPhp\Api\Lines;
-use Symfony\Component\Yaml\Yaml;
-use Symfony\Component\OptionsResolver\OptionsResolver;
 use GuzzleHttp\Client;
 use GuzzleHttp\Exception\ClientException;
-use GuzzleHttp\Exception\RequestException;
-use GuzzleHttp\Exception\ServerException;
 use CanalTP\NavitiaPhp\Exception\NavitiaException;
-use CanalTP\NavitiaPhp\Api\Coverage;
-use CanalTP\NavitiaPhp\Api\TrafficReports;
 
 class Navitia
 {
-    private $httpClient;
     /**
-     * @var string
+     * @var Client
      */
-    private $defaultCoverage;
-    private $debug;
-
-    private $coverage = null;
-    private $trafficReports = null;
-    private $coords = null;
-    private $lines = null;
+    private $httpClient;
 
     /**
      * Navitia constructor.
+     *
+     * @param Client $httpClient
      */
-    public function __construct(Client $httpClient, $coverage, $debug = false)
+    public function __construct(Client $httpClient)
     {
         $this->httpClient = $httpClient;
-        $this->defaultCoverage = $coverage;
-        $this->debug = $debug;
     }
 
     /**
-     * allow you to get different api and construct your path
+     * @param string $navitiaBaseUrl
+     * @param string $token
      *
-     * @param $apiName
-     * @return Coverage|TrafficReports|null
-     * @throws NavitiaException
+     * @return self
      */
-    public function get($apiName)
+    public static function createFromBaseUrlAndToken($navitiaBaseUrl, $token)
```

Comment:
It's a factory to avoid to create a Guzzle client before and then a Navitia client.
It simplify the steps before the first api call, so it is probably a good solution